### ISSUE TITLE
string index out of range when using Sql-editor

### ISSUE THREAD

**Issue by tudoudaren233** [dbeaver/dbeaver#23287]
### Description

when i'm using sql editor i can run my sql successfully but after i change to this same editor to another same structure of database,it jumped out this error,and i can't even change to the database which i changed from.  Idk if it's a bug or something, because i didn't met this question before this version.
![image](https://github.com/dbeaver/dbeaver/assets/59869480/32c5e2af-4a96-47af-bf7e-0c1d72237ce4)

### DBeaver Version

24.0.1

### Operating System

windows 11

**Comment by E1izabeth** [comment]
What database do you use?

**Comment by tudoudaren233** [comment]
> What database do you use?

mysql 5.6.29

**Comment by LonwoLonwo** [comment]
Could you please provide the log files? You can find instructions here: https://dbeaver.com/docs/wiki/Log-files/. Thank you.

**Comment by LonwoLonwo** [comment]
![image](https://github.com/dbeaver/dbeaver/assets/45152336/713602f4-f2d3-49ae-be29-6b3f5023b28e)

And what kind of connector do you use? MySQL or MySQL legacy? MySQL legacy is more compatible with MySQL 5.

**Comment by tudoudaren233** [comment]
Thanks for re. I'm using the mysql database connector and i will try the legacy. Here is the log.
![image](https://github.com/dbeaver/dbeaver/assets/59869480/76132632-eba5-4337-adc2-e26e089a9642)

**Comment by LonwoLonwo** [comment]
Could you please provide your database name for testing? End encoding for this database.

**Comment by tudoudaren233** [comment]
Database name is "gem", and encoding is UTF-8.

**Comment by ShadelessFox** [comment]
What is the "gem" database? Is this a fork of MySQL, a MySQL-compatible database, or the name of your database?

What driver version are you using? Can you please provide a screenshot of the "Test Connection" dialog?

**Comment by tudoudaren233** [comment]
Sorry, maybe i'v misunderstood the meaning of name of database, here is my screenshot of test connection.
![image](https://github.com/dbeaver/dbeaver/assets/59869480/016276ed-aa87-416d-8ca4-54243a658df9)

**Comment by arhayka** [comment]
@tudoudaren233  You can find the database name here, in the Database Navigator inside Databases:
<img width="321" alt="Screenshot 2024-04-11 at 18 23 15" src="https://github.com/dbeaver/dbeaver/assets/12581569/6226a9b9-746a-4612-86a4-0ed99f4e7a8c">

Could you please share it with us? Does it contain some special symbols or hieroglyphs?

**Comment by tudoudaren233** [comment]
here is the database name 
![4fdc5a282dbc9e91147f39c1c288b5e](https://github.com/dbeaver/dbeaver/assets/59869480/13b719b2-222e-44ff-a923-0a69763af163)

**Comment by Destrolaric** [comment]
@tudoudaren233 Hi is the database a original MySQL 8 or some forked version?

**Comment by tudoudaren233** [comment]
here is the mysql version, i've also tried changing the connector to 5.1.x but the problem still not fix.
![image](https://github.com/dbeaver/dbeaver/assets/59869480/391c5f9b-eaf2-4441-a87a-76aeb5188cf1)
![image](https://github.com/dbeaver/dbeaver/assets/59869480/a2a20a21-ceeb-4a67-bacd-96aa19b9a062)

**Comment by serjiokov** [comment]
Hello @tudoudaren233,  Can you connect to MySQL DB with an empty field of the database name, please provide steps on how you use SQL Editor.

**Comment by tudoudaren233** [comment]
Now i found that in new version, Although the SQL editor shows error, If i execute one of the right sql, the error will disappear.

**Comment by E1izabeth** [comment]
> Hello @tudoudaren233, Can you connect to MySQL DB with an empty field of the database name, please provide steps on how you use SQL Editor.

So does it work if you left Database field empty in the Connection Settings?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/3e24a8d4-500d-42a2-8735-c0cc8f362c2c)

Please provide the query example, which fail. And the way you execute the query. Which button do you use?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/ebc3f853-7262-4d83-aaf1-03e630cc4fbe)

**Comment by tudoudaren233** [comment]
No, it didn't work after i left the field empty, still showed this error.
![image](https://github.com/dbeaver/dbeaver/assets/59869480/918dec15-a3bb-45d3-84a0-486a51bdb044)
I use the first button to execute. But it is wired that this error did not affect my code execution, it just made everything red.
![image](https://github.com/dbeaver/dbeaver/assets/59869480/9287ff5b-4b18-4ff6-8ee1-7e6235bd7847)

**Comment by E1izabeth** [comment]
Thank you

**Comment by MashaKorax** [comment]
Unfortunately, we are unable to reproduce the issue at this time.

**Comment by tudoudaren233** [comment]
I found the way to reproduce the problem. When i rename the SQL Editor, i added some Chinese characters in the new name and then i changed to another database it will lead to this problem (i dk if other characters will also have this problems). Hope to fix it~
![image](https://github.com/user-attachments/assets/4130777f-d084-437a-b6c6-4dcb5028f71d)
![image](https://github.com/user-attachments/assets/b9e0144d-c92f-45a3-bd3b-04c2c03ebc02)

**Comment by tudoudaren233** [comment]
Here are my steps.
1. Open a new sql-editor and write some normal sql. Make sure you can run these sql correctly.
2. Save this editor and rename it with some Chinese characters like "测试sql保存报错.sql" and then close this editor.
3. Reopen this editor, and it will pop up this error.

**Comment by E1izabeth** [comment]
I followed the provided steps and everything is fine for me

**Comment by E1izabeth** [comment]
Is the issue still relevant for the latest DBeaver version?

**Comment by E1izabeth** [comment]
I suppose the issue is solved.
